### DISCUSSION TITLE
Mirth:  Oracle Java 8 to OpenJDK

### DISCUSSION THREAD

**Discussion by cldparisi** [nextgenhealthcare/connect#5716] [Q&A]
In production we have installations of Mirth using Oral Java 8.
Now we want to switch to OpenJDK, do you recommend switching to OpenJDK version 8 or the latest available? or a specific version?

Thanks

**Comment by pacmano1** [comment]
Ubuntu user here, I typically use the latest one included in the distribution.   I want to say 17 had a problem, but you might search here and the forums.    I'd guess most of my installs have 11 these day.  

Don't forget to add the extra required options in vmoptions if using 9+

**Comment by jonbartels** [comment]
You need to follow this using anything newer than Java 9 - https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_Using_Java_9_or_greater_mirth_connect_ug.html

I have MC running on Java 17 with `VMOPTIONS=-Xmx512m,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,--add-opens=java.sql.rowset/com.sun.rowset=ALL-UNNAMED`

Whatever version you choose get the latest patch release, those will have the most recent JVM cacerts.

**Comment by RichWittlerZ** [comment]
When switching from Oracle Java to OpenJDK, it's important to consider compatibility with your existing applications and any potential performance or security issues that may arise.

If you are currently using Oracle Java 8 in production and are considering a switch to OpenJDK, you may want to consider OpenJDK version 8 as the most compatible option. This version of OpenJDK is based on the same code base as Oracle Java 8, so it should be relatively easy to switch to.

However, if you need features or improvements that are not available in OpenJDK 8, or if you're looking for longer-term support, you may want to consider a later version of OpenJDK.

The latest version of OpenJDK as of my knowledge cutoff date is OpenJDK 17. This version offers several improvements over previous versions, including performance improvements, new language features, and improved security. However, you will need to make sure that your existing applications are compatible with OpenJDK 17 before making the switch.

Ultimately, the best version of OpenJDK for your organization will depend on your specific needs and requirements. It's recommended to thoroughly test any new version before making the switch in a production environment to ensure compatibility and avoid any issues that could impact your operations.

**Reply by tonygermano** [reply]
Please no ChatGPT answers

**Comment by JoeFox82** [comment]
We have all our 30+ mirth servers (3.4.2 - 4.2) running on Windows server (2012 R2 - 2022) with Adoptium openJDK 8 and have had no problems so far.

**Reply by MichaelLeeHobbs** [reply]
On the server side agree. Client-side openJDK can get a bit flaky. These days just use whatever the launcher does.

**Comment by rjkroll** [comment]
We went from Java 8 to OpenJDK 19.0.2 without any issues, other then having change some syntax or two from some custom Java things that we were doing.
